# Long-read-sequencing-README

## Brief summary

Long-read Sequencing
To date, most large-scale genetic sequencing efforts for Alzheimer's disease and related dementias have been performed using short-read DNA sequencing. Although these approaches can identify single nucleotide changes and small indels, they are not optimized to identify large structural variations or repeat expansions. Furthermore, many areas of the genome cannot be accurately sequenced with this technology, like homologous elements, highly GC-rich regions, centromeric regions, and telomeres.

Long-read sequencing enables us to generate accurate genetic sequencing data for challenging genomic regions to identify structural variants driving Alzheimer's disease and related dementias pathology. A greater understanding of the genetic architecture of the Alzheimer's disease and related dementias genome will lead to further insight into the disease and pathway mechanisms underlying them and new potential therapeutic targets for these diseases.

With this research, we will build a public resource consisting of long-read genome sequencing data from a large number of confirmed people with Alzheimer's disease and related dementias and healthy individuals. We will make both the raw and processed data publicly available to the community, along with our analysis pipeline, algorithms, and optimized DNA isolation protocols.

## Publications

XXX

## Graphical overview

XXX

## Overview of this code repo

1. XXXX
    1. Clinvar analyses INDI lines
    2. Genetic risk score analysis of INDI lines
    3. Genomic alteration analysis of INDI lines
2. XXX
   1. XXX
   2. XXX
3. XXX
   1. XXX
   2. XXX

## Contact details

For more information or questions please email:



## Full long-read sequencing Team:

The long-read sequencing team:

Adam Phillippy, NHGRI
Benedict Paten, UCSC
Bryan Traynor, NIA
Cornelis Blauwendraat, NIA
Fritz Sedlazeck, Baylor College of Medicine
Sonja Scholz, NINDS
and more





